lib/CoreBot.js

```diff
@@ -610,31 +610,48 @@ function Botkit(configuration) {
     };
 
     botkit.hears = function(keywords, events, cb) {
-        if (typeof(keywords) == 'string') {
-            keywords = [keywords];
-        }
-        if (typeof(events) == 'string') {
-            events = events.split(/\,/g);
+        var matchFn;
+        var theRule;
+
+        // Compile
+        if (typeof(keywords) == 'function') {
+            matchFn = keywords;
+            // Store first line of the fuction
+            theRule = (function strText(functionText){return functionText.substr(0, functionText.indexOf("\n"));})(keywords)
```

Comment:
There is a 120 length limit per line. Will appreciate if you don't make it a single line.